### ISSUE TITLE
Invalid sort in index for characters with diacritic signs (french locale)

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#736]
Sphinx 1.0.7. Theme sphinxdoc.

In French, A,a,à,À,Á are sorted at the same level (the same for c, C, ç, Ç and other letters with diacritic signs).
With Sphinx 1.0.7, theses letters are put at the end of the index list (for example, à and À are after all the a or A).
It seems to work when they are inside a word but not when this the very first letter:

Example of index entries (some words does not mean anything they are here for test):

```
      pair: Àfficher; çe marche
      pair: Àfficher; çb marche
      pair: Àfficher; ça larche
      pair: Àfficher; Ça marche
      single:ça fouette
      single:appalache
      single:appblache
      single:appàlache
```

The last three are correctly sorted.
The others are at the end of A and C are at the end.
In subentry of Àfficher, 'Ça marche' is before 'ça larche' which is incorrect (l is before m)

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/736
- Originally reported by: Anonymous
- Originally created at: 2011-07-16T10:24:45.311

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-23 07:57:44+00:00_

To get this correct for all locales is indeed complicated and would require a library of "strxfrm"-type functions for each locale that Sphinx supports.

**Comment by tk0miya** [comment]
Fixed by #6540.
Thank you for reporting.
